app/assets/javascripts/discourse/tests/integration/components/site-header-test.js

```diff
@@ -83,4 +83,22 @@ module("Integration | Component | site-header", function (hooks) {
     await click("header.d-header");
     assert.ok(!exists(".user-menu.revamped"));
   });
+
+  test("header's height is setting css property", async function (assert) {
+    await render(hbs`<SiteHeader />`);
+
+    function getProperty() {
+      return getComputedStyle(document.body).getPropertyValue(
+        "--header-offset"
+      );
+    }
+
+    document.querySelector(".d-header").style.height = 90 + "px";
+    await waitUntil(() => getProperty() === "90px", { timeout: 100 });
```

Comment:
Ideally I wouldn't use waitUntil but the mix of resizeObserver and css property makes it hard to do otherwise IMO.